src/analysis/calculus/times_cont_diff.lean

```diff
@@ -1626,6 +1626,22 @@ lemma times_cont_diff_within_at_snd {s : set (E × F)} {p : E × F} {n : with_to
   times_cont_diff_within_at 𝕜 n (prod.snd : E × F → F) s p :=
 times_cont_diff_snd.times_cont_diff_within_at
 
+/--
+The natural equivalence `(E × F) × G ≃ E × (F × G)` is smooth.
+-/
```

Comment:
I would like a warning here, something like `if you think you need this lemma, it is likely that you can simplify your proof by reformulating the lemma that you're applying next using the tips in Note [continuity lemma statement]`

Reply:
I have added this warning.